### ISSUE TITLE
[HOLD for payment 2023-11-02] [$500] Workspace - Archived Chat shows the edit composer for the message

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#26887]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open App > Go to Settings > Workspaces > Press New Workspace
2. Send a message to the newly created admin workspace channel
3. Press Edit Comment. It will open the edit composer for the message
4. Now, Go to Settings > Workspace > Newly created workspace > Press 3 dot from the top menu > Press Delete Workspace
5. Navigate back to the admin channel

## Expected Result:
If Chat is archived, the edit composer should be hidden because the Edit message for archived Chat is prevented from BE

## Actual Result:
Chat is archived, but the edit composer shows the message.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.64-2

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856 

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/8828c853-e8fc-42b9-ab95-243aa60da1c6

https://github.com/Expensify/App/assets/115492554/d9dfb487-8c45-4b85-b7e7-35ca6fed7975

**Expensify/Expensify Issue URL:**

**Issue reported by:** @jayeshmangwani

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1693300421650139

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0144ce6446180ed0fc</li>
        <li>Upwork Job ID: 1701302019279601664</li>
        <li>Last Price Increase: 2023-10-02</li>
<li>Automatic offers: </li>
<ul>
<li>aimane-chnaif | Reviewer | 27282551</li>
<li>bernhardoj | Contributor | 27282553</li>
<li>jayeshmangwani | Reporter | 27282554</li>
</ul></li>
    </ul>
</details>

**Comment by sakluger** [comment]
Good find!

**Comment by sakluger** [comment]
@aimane-chnaif could you please review the proposals we've received so far?

**Comment by sakluger** [comment]
@aimane-chnaif what's the next step on this one? Are we still waiting on a new package version?

**Comment by sakluger** [comment]
The new PR was just merged today, so we only need to hold for another day or two. I'll leave as daily so we can check back once it's deployed.

**Comment by sakluger** [comment]
@bernhardoj I think the Onyx issue has been solved now, are you able to verify your solution?

**Comment by sakluger** [comment]
@tgolen do you have any idea what's going on here? I'm tagging you because you reviewed [the PR](https://github.com/Expensify/react-native-onyx/pull/333#issuecomment-1722726499) that @bernhardoj said was related. If you don't know, do you have any ideas who might?

**Comment by sakluger** [comment]
@bernhardoj what is the next step here? Are more backend fixes needed, or can you continue with your proposal?

**Comment by sakluger** [comment]
Hey @aimane-chnaif can you confirm if we should move forward with @bernhardoj's proposal?

**Comment by cead22** [comment]
- It looks like the backend doesn't use the drafts key
- I'll make an internal issue so DeleteWorkspace API pushes down onyx data that removes drafts from reports
- Let's fallback on `''` instead of `undefined` in the proposed code copied below

> ```
> if (lodashGet(props.drafts, [`${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${report.parentReportID}`, props.action.reportActionID], undefined)) {
> ```

**Comment by sakluger** [comment]
I've completed payment to @jayeshmangwani and @bernhardoj. @aimane-chnaif could you please complete the BZ checklist? Once you do I can finish payments and close the issue.

**Comment by sakluger** [comment]
Thanks!
